CloverConnectorImpl.js

```diff
@@ -399,23 +409,28 @@ CloverConnectorImpl.prototype.mapDiscoveryResponse = function() {
         this.remoteMessageParser.parseMessage(message, discoveryResponse);
 
         /**  This goes in a deserialization class **/
-        var merchantInfo = new sdk.remotepay.MerchantInfo();
-        merchantInfo.setMerchantID(discoveryResponse.getMerchantId());
-        merchantInfo.setMerchantName(discoveryResponse.getMerchantName());
-        merchantInfo.setMerchantMID(discoveryResponse.getMerchantMId());
-        merchantInfo.setSupportsTipAdjust(discoveryResponse.getSupportsTipAdjust());
-        merchantInfo.setSupportsManualRefunds(discoveryResponse.getSupportsManualRefund());
+        this.merchantInfo = new sdk.remotepay.MerchantInfo();
+        this.merchantInfo.setMerchantID(discoveryResponse.getMerchantId());
+        this.merchantInfo.setMerchantName(discoveryResponse.getMerchantName());
+        this.merchantInfo.setMerchantMID(discoveryResponse.getMerchantMId());
+        this.merchantInfo.setSupportsTipAdjust(discoveryResponse.getSupportsTipAdjust());
+        this.merchantInfo.setSupportsManualRefunds(discoveryResponse.getSupportsManualRefund());
 
         var deviceInfo = new sdk.remotepay.DeviceInfo();
         deviceInfo.setName( discoveryResponse.getName() );
         deviceInfo.setSerial(discoveryResponse.getSerial());
         deviceInfo.setModel(discoveryResponse.getModel());
 
-        merchantInfo.setDeviceInfo(deviceInfo);
+        this.merchantInfo.setDeviceInfo(deviceInfo);
 
         this.deviceSupportsAckMessages = discoveryResponse.supportsAcknowledgement;
 
-        this.delegateCloverConnectorListener.onReady(merchantInfo);
+        if(!this.isReady && discoveryResponse.ready) {
```

Comment:
if the connection is made and the Mini sends a new DiscoveryResponse, it will report as onConnected...?